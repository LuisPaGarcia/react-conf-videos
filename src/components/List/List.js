import React from 'react'
import { connect } from 'react-redux'
import { compose, pure, withHandlers } from 'recompose'
import { frontPageActions } from 'redux/modules'
import Video from 'components/Videos/Video'
import { flatten } from 'ramda'
import cn from 'classnames'

import styles from './List.scss'

const ListInner = ({
  conferences, onInputChange, filterValue, isActive
}) => {
  const children = flatten(Object.keys(conferences).map((conference) => {
    const conferenceProps = conferences[conference]
    return conferenceProps.videos.map(video => <Video key={video} videoId={video} conferenceId={conference} />)
  }))
  return  (
    <div className={cn(styles.root, { [`${styles.isActive}`]: isActive })}>
      <input type="text" value={filterValue} onChange={onInputChange} className={styles.input} />
      {/* <p className={styles.results}>{children.length} {children.length === 1 ? 'Result' : 'Results'} </p> */}
      <section className={styles.results}>
        { true && children }
      </section>
    </div>
  )
}

const mapStateToProps = ({ frontPage }) => ({
  conferences: frontPage.filteredConferences,
  filterValue: frontPage.filterValue,
  isActive: frontPage.isActive
})

const mapDispatchToProps = {
  filter: frontPageActions.filter,
  setIsActive: frontPageActions.setIsActive,
}

const List = compose(
  connect(mapStateToProps, mapDispatchToProps),
  pure,
  withHandlers({
    onInputChange: ({ filter, setIsActive }) => (e) => {
      setIsActive(e.target.value !== '')
      filter(e.target.value)
    }
  })
)(ListInner)

export default List
