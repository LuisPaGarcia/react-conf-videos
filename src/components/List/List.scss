.root {
  // transition: transform 0.4s;

  // &.isActive {
    // transform: translateY(-100px);
    // transition: transform 0.4s;
  // }

  .input {
    padding: 1rem 2rem;
    background: #EFEEEE;
    font-size: 2.6rem;
    color: #404d5b;
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    width: 100%;
    border: 0;
    outline: none;
    box-sizing: border-box;
  }

  .results {
    margin-top: 2rem;
  }
}