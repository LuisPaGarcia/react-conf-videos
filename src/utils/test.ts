import { createMockStore } from 'redux-test-utils'
import { MiddlewareAPI } from 'redux'
import { mount, shallow } from 'enzyme'
import { ReactElement } from 'react'

/** epics **/

// fails test of error cb of subcribe is called
export const onError = (done:(idx: any) => any) => (error: Error) => done(false)

// mock empty store object
export const mockStore = () => { return <MiddlewareAPI<void>>{} }

/** Components **/

// shallow mount component with props
// export const shallowWithProps = <P>(Component: ReactElement<any>, props: P) => shallow(<Component {...props} />)

// shallow mount component with a store in context
export const shallowWithStore = <T>(storeP: T, component: ReactElement<any>) => {
  const store = storeP.hasOwnProperty('isActionTypeDispatched') ? storeP : createMockStore(storeP)
  return shallow<any, any>(component, { context: { store } })
}

export const mountWithStore = <T>(storeP: T, component: ReactElement<any>) => {
  const store = storeP.hasOwnProperty('isActionTypeDispatched') ? storeP : createMockStore(storeP)
  return mount<any, any>(component, { context: { store } })
}


