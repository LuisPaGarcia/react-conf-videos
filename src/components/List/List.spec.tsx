import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { mockConference, shallowWithStore } from 'utils'
import List, { ListInner } from './List'

describe('List', () => {
  describe('ListInner', () => {
    it('should render', () => {
      // arrange
      const conf = mockConference()
      const props = { conferences: {'yyy': conf} }

      // act
      const wrapper = shallow(<ListInner {...props} />)

      // assert
      expect(toJSON(wrapper)).toMatchSnapshot()
    })
  })

  describe('List', () => {
    it('should render and connect', () => {
      // arrange
      const conf = mockConference();
      const conferences = {'yyy': conf}
      const state = { frontPage: { filteredConferences: conferences }}

      // act
      const wrapper = shallowWithStore(state, <List />)

      // assert
      expect(toJSON(wrapper)).toMatchSnapshot()
      expect(wrapper.props().conferences).toEqual(conferences)
    })
  })
})