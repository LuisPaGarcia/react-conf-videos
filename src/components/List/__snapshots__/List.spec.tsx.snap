// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List List should render and connect 1`] = `
<pure(Component)
  conferences={
    Object {
      "yyy": Object {
        "date": "XX/YY/ZZZ",
        "playlist": "day 1",
        "title": "fake conf title",
        "videos": Array [
          "aaa",
          "bbb",
          "ccc",
        ],
        "website": "fake url",
      },
    }
  }
  dispatch={[Function]}
/>
`;

exports[`List ListInner should render 1`] = `
<div
  className="root"
>
  <Connect(pure(withStateHandlers(withHandlers(Component)))) />
  <p
    className="resultsCount"
  >
    <span
      className="resultsNumber"
    >
       
      3
       
    </span>
     
    videos
    <span
      className="resultsNumber"
    >
       
      1
       
    </span>
     
    conference
  </p>
  <section
    className="results"
  >
    <Connect(pure(withStateHandlers(Component)))
      conferenceId="yyy"
      key="aaa"
      videoId="aaa"
    />
    <Connect(pure(withStateHandlers(Component)))
      conferenceId="yyy"
      key="bbb"
      videoId="bbb"
    />
    <Connect(pure(withStateHandlers(Component)))
      conferenceId="yyy"
      key="ccc"
      videoId="ccc"
    />
  </section>
</div>
`;
